src/main/java/com/github/paweladamski/httpclientmock/HttpClientResponseBuilder.java

```diff
@@ -173,7 +180,7 @@ public HttpClientResponseBuilder doReturnXML(String response) {
      * @return response builder
      */
     public HttpClientResponseBuilder doReturnXML(String response, Charset charset) {
-        return doReturn(response, charset).withHeader("Content-type", APPLICATION_XML.toString());
+        return doReturn(response, charset, APPLICATION_XML).withHeader("Content-type", APPLICATION_XML.toString());
```

Comment:
It looks a bit confusing that we use APPLICATION_XML twice here. 
I would prefer to move setting "Content-type" header to `StringResponse#getResponse`.

Reply:
How to identify if we need to set XML or JSON content type in `StringResponse#getResponse` ?